### ISSUE TITLE
🐞 CLI args not parsed when secret env is not defined

### ISSUE THREAD

**Issue by skycaptain** [dagger/dagger#7181]
### What is the issue?

I have a module, where using credentials is optional, e.g.

```python
import dagger
from dagger import function, object_type
 
@object_type
class Test:
    username: str | None = None
    token: dagger.Secret | None = None
 
    name: str | None = None
 
    @function
    def call(self) -> str:
        return f"Hello, {self.name}!"
```

When I call this module like this, I get a valid response:

```bash
$ MY_TOKEN="very_secret_variable" dagger call -m test --token=env:MY_TOKEN --name "World" call
Hello, World!
```

However, when token is missing, it seems that the CLI is stopping parsing the other arguments:

```bash
$ dagger call -m test --token=env:MY_TOKEN --name "World" call
Hello, None!
```

### Dagger version

dagger v0.11.1 ([registry.dagger.io/engine](http://registry.dagger.io/engine)) darwin/arm64

### Steps to reproduce

See above.

### Log output

*No response*

**Comment by helderco** [comment]
That's an interesting find, thanks for reporting! 🤔

**Comment by helderco** [comment]
This was fixed in [v0.11.5](https://github.com/dagger/dagger/releases/tag/v0.11.5). Expected output:

```console
$ dagger call -m test --token=env:MY_TOKEN --name "World" call

Error: failed to get value for argument "token": secret env var not found: "MY_..."
Run 'dagger call --help' for usage.
```
